aws/resource_aws_chime_voice_connector_termination.go

```diff
@@ -0,0 +1,191 @@
+		},
+
+		Schema: map[string]*schema.Schema{
+			"calling_regions": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringLenBetween(2, 2),
+				},
+			},
+			"cidr_allow_list": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.IsCIDRNetwork(27, 32),
+				},
+			},
+			"cps_limit": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1,
+				ValidateFunc: validation.IntAtMost(1),
+			},
+			"default_phone_number": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^\+?[1-9]\d{1,14}$`), "must match ^\\+?[1-9]\\d{1,14}$"),
+			},
+			"disabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"voice_connector_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsChimeVoiceConnectorTerminationPut(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).chimeconn
+
+	vcId := d.Get("voice_connector_id").(string)
+
+	input := &chime.PutVoiceConnectorTerminationInput{
+		VoiceConnectorId: aws.String(vcId),
+	}
+
+	termination := &chime.Termination{
+		CidrAllowedList: expandStringList(d.Get("cidr_allow_list").([]interface{})),
+		CallingRegions:  expandStringList(d.Get("calling_regions").([]interface{})),
+	}
+
+	if v, ok := d.GetOk("disabled"); ok {
+		termination.Disabled = aws.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("cps_limit"); ok {
+		termination.CpsLimit = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("default_phone_number"); ok {
+		termination.DefaultPhoneNumber = aws.String(v.(string))
+	}
+
+	input.Termination = termination
+
+	if _, err := conn.PutVoiceConnectorTerminationWithContext(ctx, input); err != nil {
+		return diag.Errorf("error creating Chime Voice Connector (%s) termination: %s", vcId, err)
+	}
+
+	d.SetId(fmt.Sprintf("termination-%s", vcId))
+
+	return resourceAwsChimeVoiceConnectorTerminationRead(ctx, d, meta)
+}
+
+func resourceAwsChimeVoiceConnectorTerminationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).chimeconn
+
+	vcId := d.Get("voice_connector_id").(string)
+	input := &chime.GetVoiceConnectorTerminationInput{
+		VoiceConnectorId: aws.String(vcId),
+	}
+
+	resp, err := conn.GetVoiceConnectorTerminationWithContext(ctx, input)
+	if !d.IsNewResource() && isAWSErr(err, chime.ErrCodeNotFoundException, "") {
+		log.Printf("[WARN] error getting Chime Voice Connector (%s) termination: %s", vcId, err)
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil || resp.Termination == nil {
+		return diag.Errorf("error getting Chime Voice Connector (%s) termination: %s", vcId, err)
+	}
```

Comment:
we can re-use the ID once the change is made above e.g.
```suggestion
	input := &chime.GetVoiceConnectorTerminationInput{
		VoiceConnectorId: aws.String(d.Id()),
	}

	resp, err := conn.GetVoiceConnectorTerminationWithContext(ctx, input)
	if !d.IsNewResource() && isAWSErr(err, chime.ErrCodeNotFoundException, "") {
		log.Printf("[WARN] Chime Voice Connector Termination (%s) not found, removing from state: %s", d.Id(), err)
		d.SetId("")
		return nil
	}

	if err != nil {
	        return diag.Errorf("error getting Chime Voice Connector Termination (%s)": %s, d.Id(), err)
	}
	
	if resp == nil || resp.Termination == nil {
	          return diag.Errorf("error getting Chime Voice Connector Termination (%s): empty response", d.Id())
	}
```